crates/sui-tls/src/lib.rs

```diff
@@ -51,6 +55,45 @@ mod tests {
             .unwrap();
     }
 
+    #[test]
+    fn verify_server_cert() {
+        let mut rng = rand::thread_rng();
+        let allowed = Ed25519KeyPair::generate(&mut rng);
+        let disallowed = Ed25519KeyPair::generate(&mut rng);
+        let allowed_public_key = allowed.public().to_owned();
+        let random_cert_bob =
+            SelfSignedCertificate::new(allowed.private(), SUI_VALIDATOR_SERVER_NAME);
+        let random_cert_alice =
+            SelfSignedCertificate::new(disallowed.private(), SUI_VALIDATOR_SERVER_NAME);
+
+        let verifier =
+            ServerCertVerifier::new(allowed_public_key, SUI_VALIDATOR_SERVER_NAME.to_string());
+
+        // The bob passes validation
+        verifier
+            .verify_server_cert(
+                &random_cert_bob.rustls_certificate(),
+                &[],
+                &rustls::ServerName::try_from("example.com").unwrap(),
+                &mut Vec::<&[u8]>::new().iter().cloned(),
+                &[],
+                std::time::SystemTime::now(),
+            )
+            .unwrap();
+
+        // The alice does not pass validation
+        verifier
+            .verify_server_cert(
+                &random_cert_alice.rustls_certificate(),
+                &[],
+                &rustls::ServerName::try_from("example.com").unwrap(),
+                &mut Vec::<&[u8]>::new().iter().cloned(),
+                &[],
+                std::time::SystemTime::now(),
+            )
+            .unwrap_err();
```

Comment:
I know that is obvious but can we confirm that the error we get back is the expected one , `is not the expected server public key...`  to make sure that we fail for the right reason?

Reply:
Sure.